roles/kubernetes/preinstall/handlers/main.yml

```diff
@@ -32,4 +36,33 @@
   service:
     name: kubelet
     state: restarted
-  when: "{{ inventory_hostname in groups['kube-master'] and not dns_early|bool }}"
+  notify:
+    - Preinstall | kube-controller configured
+    - Preinstall | stop controller
+    - Preinstall | pause for controller
+    - Preinstall | restart controller
+  when: inventory_hostname in groups['kube-master'] and not dns_early|bool
+
+- name: Preinstall | kube-controller configured
+  stat: path="{{ kube_manifest_dir }}/kube-controller-manager.manifest"
+  register: kube_controller_set
+  when: inventory_hostname in groups['kube-master'] and dns_mode != 'none' and resolvconf_mode == 'host_resolvconf'
+
+- name: Preinstall | stop controller
```

Comment:
could you please rework with just sending kill -TERM to the process? It shall proceed OK even though running in the container.